pennylane/templates/embeddings.py

```diff
@@ -17,10 +17,82 @@
 used at the beginning of a circuit.
 """
 #pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
-from collections.abc import Iterable
+from pennylane import numpy as np
+from pennylane.qnode import Variable, QuantumFunctionError
 from pennylane.ops import RX, RY, RZ, BasisState, Squeezing, Displacement, QubitStateVector
-from pennylane.variable import Variable
-import numpy as np
+from pennylane.templates.utils import (_check_shape, _check_no_variable, _check_wires,
+                                       _check_hyperp_is_in_options, _check_type)
+
+
+def AmplitudeEmbedding(features, wires, pad=False, normalize=False, c=0.):
+    r"""Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits.
+
+    If the total number of features to embed is less than the :math:`2^n` available amplitudes,
+    non-informative constants (zeros) can be padded to ``features``. To enable this, the argument
+    ``pad`` should be set to ``True``.
+
+    The L2-norm of ``features`` must be one. By default, AmplitudeEmbedding expects a normalized
+    feature vector. The argument ``normalize`` can be set to ``True`` to automatically normalize it.
+
+    .. note::
+
+        AmplitudeEmbedding uses PennyLane's :class:`~pennylane.ops.QubitStateVector` and only works in conjunction with
+        devices that support this operation.
+
+    Args:
+        features (array): input array of shape ``(2**n,)``
+        wires (Sequence[int] or int): int or sequence of qubit indices that the template acts on
+
+    Keyword Args:
+        pad (Boolean): controls the activation of the padding option
+        normalize (Boolean): controls the activation of automatic normalization
+        c (float or complex): used as the constant for padding
+
+    Raises:
+        QuantumFunctionError if inputs do not have the correct format.
+    """
+
+    #############
+    # Input checks
+    _check_no_variable([pad, normalize], ['pad', 'normalize'])
```

Comment:
Better throw a ValueError? I was not sure, I thought everything a QNode has problems with gets a QuantumFunctionError. But I suspected that this sweeping use was not the best idea.

Reply:
Changed to ValueErrors

Reply:
Actually that is not what I meant -- if you give no `msg` argument here the check will raise the defaul error message which is something like `... QNode ...`. I just wanted to know if this is appropriate here to speak of the QNode. Just wondered if this might be used outside of this context.

Reply:
I imrpoved error messages, trying now to always use template-defined messages except in some very generic cases.